include/zephyr/input/input_touch.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2024 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_TOUCH_H_
+#define ZEPHYR_INCLUDE_TOUCH_H_
+
+/**
+ * @brief Touch Events API
+ * @defgroup touch_events Touchscreen Event Report API
+ * @since 3.7
+ * @version 0.1.0
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/input/input.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Common touchscreen config.
+ *
+ * This structure **must** be placed first in the driver's config structure.
+ */
+struct input_touchscreen_common_config {
+	uint32_t screen_width;
+	uint32_t screen_height;
+	bool inverted_x;
+	bool inverted_y;
+	bool swapped_x_y;
+};
+
+/**
+ * @brief Initialize common touchscreen config from devicetree
+ *
+ * @param node_id The devicetree node identifier.
+ */
+#define INPUT_TOUCH_DT_COMMON_CONFIG_INIT(node_id)			\
+	{								\
+		.screen_width = DT_PROP(node_id, screen_width),		\
+		.screen_height = DT_PROP(node_id, screen_height),	\
+		.inverted_x = DT_PROP(node_id, inverted_x),		\
+		.inverted_y = DT_PROP(node_id, inverted_y),		\
+		.swapped_x_y = DT_PROP(node_id, swapped_x_y)		\
+	}
+
+/**
+ * @brief Initialize common touchscreen config from devicetree instance.
+ *
+ * @param inst Instance.
+ */
+#define INPUT_TOUCH_DT_INST_COMMON_CONFIG_INIT(inst) \
+	INPUT_TOUCH_DT_COMMON_CONFIG_INIT(DT_DRV_INST(inst))
+
+/**
+ * @brief Common utility for reporting touchscreen position events.
+ */
+void input_touchscreen_report_pos(const struct device *dev, uint32_t x, uint32_t y);
+
+#ifdef __cplusplus
```

Comment:
Documentation for the parameters, maybe explaining if the x and y coordinates passed in refer to the coordinates "as reported by the controller" or something. I have found that coordinates swapping is quite tricky to keep track of when following it through a codebase. The more explicit this sort of stuff is stated the easier it is to spot any logic errors imo.

Reply:
Added